tests/validators/string.test.ts

```diff
@@ -96,4 +96,208 @@ describe('StringValidator', () => {
+					);
+				});
+			});
+		});
+
+		describe('uuid', () => {
+			const uuid5s = ['2a7ff881-2944-55ae-94b0-b2ed34432297'];
+			const uuid4s = ['450d6a23-9e6f-45d9-9d5a-fd4f6e014f16'];
+			const uuid3s = ['2962d7f2-92f2-3105-8606-2234808bdfc8'];
+			const uuid1s = ['c310deb0-9785-11ec-8e65-592cb74aa664'];
+			const nullUuid = '00000000-0000-0000-0000-000000000000';
+
+			describe('uuid5', () => {
+				const uuid5Predicate = s.string.uuid({ version: 5 });
+
+				test.each(uuid5s)('GIVEN %s THEN returns given value', (input) => {
+					expect(uuid5Predicate.parse(input)).toBe(input);
+				});
+
+				test.each(['1b671a64-40d5-491e-99b0-da01ff1f3342'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => uuid5Predicate.parse(input)).toThrow(
+						new ConstraintError('s.string.uuid', 'Invalid string format', input, 'expected UUID v5')
+					);
+				});
+			});
+
+			describe('uuid4', () => {
+				const uuid4Predicate = s.string.uuid({ version: 4 });
+
+				test.each(uuid4s)('GIVEN %s THEN returns given value', (input) => {
+					expect(uuid4Predicate.parse(input)).toBe(input);
+				});
+
+				test.each(['6e8bc430-9a1b-4f7f-b7a5', '6e8bc430-9a1b-4f7f-b7a5', '450d6a23-9e6f-35d9-9d5a-fd4f6e014f16'])(
+					'GIVEN %s THEN throws a ConstraintError',
+					(input) => {
+						expect(() => uuid4Predicate.parse(input)).toThrow(
+							new ConstraintError('s.string.uuid', 'Invalid string format', input, 'expected UUID v4')
+						);
+					}
+				);
+
+				test('Given no version then match v4', () => {
+					const defaultPredicate = s.string.uuid();
+					const input = 'a0d32a64-3843-4475-90f3-7ce5b74168d8';
+					expect(uuid4Predicate.parse(input)).toStrictEqual(defaultPredicate.parse(input));
+				});
+			});
+
+			describe('uuid3', () => {
+				const uuid3Predicate = s.string.uuid({ version: 3 });
+
+				test.each(uuid3s)('GIVEN %s THEN returns given value', (input) => {
+					expect(uuid3Predicate.parse(input)).toBe(input);
+				});
+
+				test.each(['6e8bc430-9a1b-4f7f-b7a5', '450d6a23-9e6f-15d9-9d5a-fd4f6e014f16'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => uuid3Predicate.parse(input)).toThrow(
+						new ConstraintError('s.string.uuid', 'Invalid string format', input, 'expected UUID v3')
+					);
+				});
+			});
+
+			describe('with version range', () => {
+				const uuidRangePredicate = s.string.uuid({ version: '1-4' });
+
+				test.each([...uuid1s, ...uuid3s, ...uuid3s])('GIVEN %s THEN returns given value', (input) => {
+					expect(uuidRangePredicate.parse(input)).toBe(input);
+				});
+
+				test.each(uuid5s)('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => uuidRangePredicate.parse(input)).toThrow(
+						new ConstraintError('s.string.uuid', 'Invalid string format', input, `expected UUID in range 1-4`)
+					);
+				});
+			});
+
+			test('Given NIL UUID THEN return NIL UUID', () => {
```

Comment:
This should be inside a `nullable` describe.

Reply:
Also don't name it `NIL UUID`, and the words `GIVEN` and `THEN` should be always uppercase. `return` should always be `returns`.

Syntax is as it follows: `GIVEN <given> THEN <returns|throws|rejects> <result>`.

Reply:
https://github.com/uuidjs/uuid#uuidnil

It's name is `NIL` that's why I used `NIL`. So should I use `null` instead?

Reply:
Yes, `NIL` is wording from other languages, here in JavaScript we use `null`.